esquery.js

```diff
@@ -47,196 +46,267 @@ function getPath(obj, key) {
  * @param {?external:AST} node
  * @param {?external:AST} ancestor
  * @param {string[]} path
+ * @param {Integer} fromPathIndex
  * @returns {boolean}
  */
-function inPath(node, ancestor, path) {
-    if (path.length === 0) { return node === ancestor; }
-    if (ancestor == null) { return false; }
-    const field = ancestor[path[0]];
-    const remainingPath = path.slice(1);
-    if (Array.isArray(field)) {
-        for (const component of field) {
-            if (inPath(node, component, remainingPath)) { return true; }
+function inPath(node, ancestor, path, fromPathIndex) {
+    let current = ancestor;
+    for (let i = fromPathIndex; i < path.length; i++) {
+        if (current == null) {
+            return false;
         }
-        return false;
-    } else {
-        return inPath(node, field, remainingPath);
+        const field = current[path[i]];
+        if (Array.isArray(field)) {
+            for (let j = 0; j < field.length; j++) {
```

Comment:
Style nit: avoid having both an `i` and `j` in scope at the same time. Let's just rename to `k`.

```suggestion
            for (let k = 0; k < field.length; ++k) {
```